In [1]:
library(ggplot2)
options(repr.plot.height=4,repr.plot.width=6)

Cargar los datos en un dataframe llamado: airbnb

In [2]:
airbnb<-read.csv('data//airbnb.csv',sep = ',')

Mostrar las primeras 6 filas del dataframe

In [3]:
head(airbnb)

Zipcode Neighbourhood.Cleansed Property.Type Room.Type       Accommodates
1 28004   Universidad            Apartment     Private room    2           
2 28004   Universidad            Apartment     Entire home/apt 6           
3 28004   Universidad            Apartment     Entire home/apt 3           
4 28004   Universidad            Loft          Entire home/apt 3           
5 28015   Universidad            Apartment     Entire home/apt 5           
6 28004   Universidad            Apartment     Entire home/apt 2           
  Bathrooms Bedrooms Beds Bed.Type
1 2         1        1    Real Bed
2 1         3        5    Real Bed
3 1         2        2    Real Bed
4 2         1        1    Real Bed
5 1         1        1    Real Bed
6 1         0        1    Real Bed
  Amenities                                                                                                                                                                                                                                                                                                                                                       
1 TV,Wireless Internet,Kitchen,Pets allowed,Pets live on this property,Buzzer/wireless intercom,Heating,Suitable for events,Washer,First aid kit,Essentials,Lock on bedroom door,Iron                                                                                                                                                                             
2 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Indoor fireplace,Heating,Family/kid friendly,Washer,Dryer,Smoke detector,Carbon monoxide detector,Essentials,Shampoo                                                                                                                                                                                     
3 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Doorman,Heating,Family/kid friendly,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace                                                                                                                                                                                          
4 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Pets allowed,Breakfast,Elevator in building,Indoor fireplace,Buzzer/wireless intercom,Heating,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace,translation missing: en.hosting_amenity_49,translation missing: en.hosting_amenity_50                                          
5 TV,Wireless Internet,Air conditioning,Kitchen,Smoking allowed,Pets allowed,Elevator in building,Heating,Family/kid friendly,Washer,Essentials,Iron                                                                                                                                                                                                              
6 TV,Cable TV,Internet,Wireless Internet,Air conditioning,Wheelchair accessible,Kitchen,Doorman,Elevator in building,Buzzer/wireless intercom,Heating,Washer,Smoke detector,Carbon monoxide detector,First aid kit,Safety card,Fire extinguisher,Essentials,Shampoo,24-hour check-in,Hangers,Hair dryer,Iron,Laptop friendly workspace,Self Check-In,Doorman Entry
  Square.Feet Price Review.Scores.Rating
1  NA         35     NA                 
2  NA         92     96                 
3  NA         55     91                 
4  NA         60    100                 
5 538         75     90                 
6  NA         55     95

Renombrar las columnas de la siguiente forma:

| Nombre original | Nuevo nombre |
| - | - |
| Zipcode | CodigoPostal |
| Neighbourhood.Cleansed | Barrio |
| Property.Type	| TipoPropiedad |
| Room.Type | TipoAlquiler |
| Accommodates | MaxOcupantes |
| Bathrooms	| NumBanyos |
| Bedrooms	| NumDormitorios |
| Beds	|  NumCamas |
| Bed.Type	| TipoCama |
| Amenities | Comodidades |
| Square.Feet | PiesCuadrados |
| Price	| Precio |
| Review.Scores.Rating | Puntuacion |

In [4]:
newnames<-c("CodigoPostal","Barrio","TipoPropiedad","TipoAlquiler","MaxOcupantes","NumBanyos",
"NumDormitorios","NumCamas","TipoCama","Comodidades","PiesCuadrados","Precio","Puntuacion")


colmames(airbnb)

In [7]:
colnames(airbnb)
colnames(airbnb)<-newnames
colnames(airbnb)

[1] "Zipcode"                "Neighbourhood.Cleansed" "Property.Type"         
 [4] "Room.Type"              "Accommodates"           "Bathrooms"             
 [7] "Bedrooms"               "Beds"                   "Bed.Type"              
[10] "Amenities"              "Square.Feet"            "Price"                 
[13] "Review.Scores.Rating"

[1] "CodigoPostal"   "Barrio"         "TipoPropiedad"  "TipoAlquiler"  
 [5] "MaxOcupantes"   "NumBanyos"      "NumDormitorios" "NumCamas"      
 [9] "TipoCama"       "Comodidades"    "PiesCuadrados"  "Precio"        
[13] "Puntuacion"

Crea una nueva columna llamada MetrosCuadrados a partir de la columna PiesCuadrados.

Ayuda: 1 pie cuadrado son 0,092903 metros cuadrdados

In [9]:
airbnb$MetrosCuadrados<-airbnb$PiesCuadrados*0.092903 

In [11]:
head(airbnb)

CodigoPostal Barrio      TipoPropiedad TipoAlquiler    MaxOcupantes NumBanyos
1 28004        Universidad Apartment     Private room    2            2        
2 28004        Universidad Apartment     Entire home/apt 6            1        
3 28004        Universidad Apartment     Entire home/apt 3            1        
4 28004        Universidad Loft          Entire home/apt 3            2        
5 28015        Universidad Apartment     Entire home/apt 5            1        
6 28004        Universidad Apartment     Entire home/apt 2            1        
  NumDormitorios NumCamas TipoCama
1 1              1        Real Bed
2 3              5        Real Bed
3 2              2        Real Bed
4 1              1        Real Bed
5 1              1        Real Bed
6 0              1        Real Bed
  Comodidades                                                                                                                                                                                                                                                                                                                                                     
1 TV,Wireless Internet,Kitchen,Pets allowed,Pets live on this property,Buzzer/wireless intercom,Heating,Suitable for events,Washer,First aid kit,Essentials,Lock on bedroom door,Iron                                                                                                                                                                             
2 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Indoor fireplace,Heating,Family/kid friendly,Washer,Dryer,Smoke detector,Carbon monoxide detector,Essentials,Shampoo                                                                                                                                                                                     
3 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Doorman,Heating,Family/kid friendly,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace                                                                                                                                                                                          
4 TV,Internet,Wireless Internet,Air conditioning,Kitchen,Pets allowed,Breakfast,Elevator in building,Indoor fireplace,Buzzer/wireless intercom,Heating,Washer,Essentials,Shampoo,Hangers,Hair dryer,Iron,Laptop friendly workspace,translation missing: en.hosting_amenity_49,translation missing: en.hosting_amenity_50                                          
5 TV,Wireless Internet,Air conditioning,Kitchen,Smoking allowed,Pets allowed,Elevator in building,Heating,Family/kid friendly,Washer,Essentials,Iron                                                                                                                                                                                                              
6 TV,Cable TV,Internet,Wireless Internet,Air conditioning,Wheelchair accessible,Kitchen,Doorman,Elevator in building,Buzzer/wireless intercom,Heating,Washer,Smoke detector,Carbon monoxide detector,First aid kit,Safety card,Fire extinguisher,Essentials,Shampoo,24-hour check-in,Hangers,Hair dryer,Iron,Laptop friendly workspace,Self Check-In,Doorman Entry
  PiesCuadrados Precio Puntuacion MetrosCuadrados
1  NA           35      NA              NA       
2  NA           92      96              NA       
3  NA           55      91              NA       
4  NA           60     100              NA       
5 538           75      90        49.98181       
6  NA           55      95              NA

Miremos el código postal. Es una variable con entradas erroneas.
Hay valores como '', '-'  y '28' que deberían ser considerados como NA.
Así mismo también debería ser NA todos los que no compiencen por 28, ya que estamos con códigos postales de Madrid

El código postal 28002, 28004 y 28051 tienen entradas repetidas. 
Por ejemplo las entradas 28002\n20882 deberían ir dnetro de 28002

El codigo 2804 debería ser 28004, 2805 deberia ser 28005 y 2815 juncto con 2815 debería ser 28015

Limpia los datos de la columna Codigo Postal

In [14]:
str(airbnb$CodigoPostal )
levels(airbnb$CodigoPostal )

 Factor w/ 77 levels "","-","20013",..: 15 15 15 15 26 15 15 15 26 15 ...


[1] ""             "-"            "20013"        "20126"        "2015"        
 [6] "25008"        "27004"        "27013"        "28"           "28001"       
[11] "280013"       "28002"        "28002\n28002" "28003"        "28004"       
[16] "28005"        "28006"        "28007"        "28008"        "28009"       
[21] "28010"        "28011"        "28012"        "28013"        "28014"       
[26] "28015"        "28016"        "28017"        "28018"        "28019"       
[31] "28020"        "28021"        "28022"        "28023"        "28024"       
[36] "28025"        "28026"        "28027"        "28028"        "28029"       
[41] "28030"        "28031"        "28032"        "28033"        "28034"       
[46] "28035"        "28036"        "28037"        "28038"        "28039"       
[51] "2804"         "28040"        "28041"        "28042"        "28043"       
[56] "28044"        "28045"        "28046"        "28047"        "28048"       
[61] "28049"        "2805"         "28050"        "28051"        "28051\n28051"
[66] "28052"        "28053"        "28054"        "28055"        "28056"       
[71] "28058"        "28060"        "28094"        "28105"        "2815"        
[76] "28850"        "Madrid 28004"

In [15]:
airbnb$CodigoPostal<-as.character(airbnb$CodigoPostal)
str(airbnb$CodigoPostal )

 chr [1:13207] "28004" "28004" "28004" "28004" "28015" "28004" "28004" ...


Una vez limpios los datos ¿Cuales son los códigos postales que tenemos? 

In [16]:
airbnb$CodigoPostal[airbnb$CodigoPostal == ""|airbnb$CodigoPostal == "-"] <- NA

In [23]:
airbnb$CodigoPostal

[1] "28004"        "28004"        "28004"        "28004"       
    [5] "28015"        "28004"        "28004"        "28004"       
    [9] "28015"        "28004"        "28004"        "28004"       
   [13] "28013"        "28005"        "28013"        "28013"       
   [17] "28013"        "28012"        "28013"        "28013"       
   [21] "28013"        "28005"        "28013"        "28012"       
   [25] "28012"        "28013"        "28014"        "28013"       
   [29] "28012"        "28005"        "28005"        "28005"       
   [33] "28005"        "28012"        "28005"        "28005"       
   [37] "28045"        "28005"        "28005"        "28005"       
   [41] "28005"        "28005"        "28013"        "28013"       
   [45] "28013"        "28013"        "28013"        "28012"       
   [49] "28013"        "28012"        "28013"        "28013"       
   [53] "28013"        "28012"        "28013"        "28013"       
   [57] "28012"        "27013"        NA             "28045"       
   [61] "28045"        "28045"        "28045"        "28045"       
   [65] "28045"        "28045"        "28045"        "28045"       
   [69] "28045"        "28045"        "28045"        "28012"       
   [73] "28045"        "28045"        "28045"        "28045"       
   [77] "28012"        "28012"        "28012"        "28012"       
   [81] "28005"        "28012"        NA             "28005"       
   [85] "28012"        "28012"        "28012"        "28012"       
   [89] "28012"        "28005"        "28012"        "28005"       
   [93] "28005"        "28012"        "28005"        "28012"       
   [97] "28012"        "28005"        "28005"        "28005"       
  [101] "28012"        "28012"        "28005"        "28012"       
  [105] "28014"        "28014"        "28014"        "28014"       
  [109] "28012"        "28014"        "28014"        "28014"       
  [113] "28012"        "28014"        "28014"        "28014"       
  [117] "28014"        "28014"        "28012"        "28012"       
  [121] "28014"        "28012"        "28012"        "28005"       
  [125] "28012"        "28012"        "28012"        "28012"       
  [129] "28012"        "28005"        "28005"        NA            
  [133] "28012"        NA             "28012"        "28012"       
  [137] "28012"        "28005"        "28012"        "28012"       
  [141] "28014"        "28014"        "28045"        "28045"       
  [145] "28045"        "28045"        "28007"        "28007"       
  [149] "28007"        "28007"        "28007"        "28007"       
  [153] "28028"        "28009"        "28009"        "28009"       
  [157] "28009"        "28014"        "28014"        "28007"       
  [161] "28005"        "28013"        "28005"        "28005"       
  [165] "28013"        "28013"        "28013"        "28005"       
  [169] "28013"        "28013"        "28005"        "28005"       
  [173] "28005"        "28013"        "28013"        "28013"       
  [177] "28005"        "28013"        "28005"        "28013"       
  [181] "28013"        "28005"        "28013"        "28005"       
  [185] "28013"        "28013"        "28005"        "28005"       
  [189] "28013"        "28013"        "28005"        "28013"       
  [193] "28005"        "28013"        "28005"        "28013"       
  [197] "28005"        "28005"        "28005"        "28005"       
  [201] "28005"        "28013"        "28005"        "28005"       
  [205] "28005"        "28013"        "28013"        "28012"       
  [209] "28005"        "28012"        "28005"        "28012"       
  [213] "28005"        "28012"        "28012"        "28005"       
  [217] "28012"        "28005"        "28012"        "28012"       
  [221] "28012"        "28012"        "28012"        "28012"       
  [225] "28012"        "28012"        "28012"        "28013"       
  [229] "28013"        "28014"        "28012"        "28012"       
  [233] "28012"        "28005"        "28005"        "28005"

In [24]:
airbnb$CodigoPostal<-as.factor(airbnb$CodigoPostal)
airbnb$CodigoPostal

[1] 28004        28004        28004        28004        28015       
    [6] 28004        28004        28004        28015        28004       
   [11] 28004        28004        28013        28005        28013       
   [16] 28013        28013        28012        28013        28013       
   [21] 28013        28005        28013        28012        28012       
   [26] 28013        28014        28013        28012        28005       
   [31] 28005        28005        28005        28012        28005       
   [36] 28005        28045        28005        28005        28005       
   [41] 28005        28005        28013        28013        28013       
   [46] 28013        28013        28012        28013        28012       
   [51] 28013        28013        28013        28012        28013       
   [56] 28013        28012        27013        <NA>         28045       
   [61] 28045        28045        28045        28045        28045       
   [66] 28045        28045        28045        28045        28045       
   [71] 28045        28012        28045        28045        28045       
   [76] 28045        28012        28012        28012        28012       
   [81] 28005        28012        <NA>         28005        28012       
   [86] 28012        28012        28012        28012        28005       
   [91] 28012        28005        28005        28012        28005       
   [96] 28012        28012        28005        28005        28005       
  [101] 28012        28012        28005        28012        28014       
  [106] 28014        28014        28014        28012        28014       
  [111] 28014        28014        28012        28014        28014       
  [116] 28014        28014        28014        28012        28012       
  [121] 28014        28012        28012        28005        28012       
  [126] 28012        28012        28012        28012        28005       
  [131] 28005        <NA>         28012        <NA>         28012       
  [136] 28012        28012        28005        28012        28012       
  [141] 28014        28014        28045        28045        28045       
  [146] 28045        28007        28007        28007        28007       
  [151] 28007        28007        28028        28009        28009       
  [156] 28009        28009        28014        28014        28007       
  [161] 28005        28013        28005        28005        28013       
  [166] 28013        28013        28005        28013        28013       
  [171] 28005        28005        28005        28013        28013       
  [176] 28013        28005        28013        28005        28013       
  [181] 28013        28005        28013        28005        28013       
  [186] 28013        28005        28005        28013        28013       
  [191] 28005        28013        28005        28013        28005       
  [196] 28013        28005        28005        28005        28005       
  [201] 28005        28013        28005        28005        28005       
  [206] 28013        28013        28012        28005        28012       
  [211] 28005        28012        28005        28012        28012       
  [216] 28005        28012        28005        28012        28012       
  [221] 28012        28012        28012        28012        28012       
  [226] 28012        28012        28013        28013        28014       
  [231] 28012        28012        28012        28005        28005       
  [236] 28005        28012        28012        28005        28012       
  [241] 28012        28012        28012        28012        28012       
  [246] 28012        28014        28014        28014        28014       
  [251] 28012        28014        28014        28014        28014       
  [256] 28004        28004        28004        28004        28004       
  [261] 28004        28004        28004        28004        28004       
  [266] 28001        28001        28006        28001        28001       
  [271] 28001        28001        28001        28001        28

In [28]:
airbnb$CodigoPostal<-as.character(airbnb$CodigoPostal)
str(airbnb$CodigoPostal )

 chr [1:13207] "28004" "28004" "28004" "28004" "28015" "28004" "28004" ...


In [2]:
sapply(strsplit(airbnb$CodigoPostal[grep("\n", airbnb$CodigoPostal, perl=TRUE, value=FALSE)], split = "\n"), "[", 1)


ERROR: Error in strsplit(airbnb$CodigoPostal[grep("\n", airbnb$CodigoPostal, : object 'airbnb' not found


¿Cuales son los 5 códigos postales con más entradas? ¿Y con menos? ¿Cuantas entradas tienen?

¿Cuales son los barrios que hay en el código postal 28012?

¿Cuantas entradas hay en cada uno de esos barrios para el codigo postao 28012? Asumiendo que el identificador de Barrio sea correcto, ¿es fiable la columna de código postal?

¿Cuantos barrios hay en todo el dataset airbnb? ¿Cuales son?

¿Cuales son los 5 barrios que tienen mayor número entradas?

¿Cuantos Tipos de Alquiler diferentes hay? ¿Cuales son? ¿Cuantas entradas en el dataframe hay por cada tipo?

Muestra el diagrama de cajas del precio para cada uno de los diferentes Tipos de Alquiler

Cual es el precio medio de alquiler medio de cada uno, la diferencia que hay ¿es estadísticamente significativa?
¿Con que test lo comprobarías?

Filtra el dataframe cuyos tipo de alquiler sea  'Entire home/apt' y guardalo en un dataframe llamado 
*airbnb_entire*.
Estas serán las entradas que tienen un alquiler del piso completo.

¿Cuales son los 5 barrios que tienen un mayor número de apartamentos enteros en alquiler?
Nota: Mirar solo en airbnb_entire

¿Cuales son los 5 barrios que tienen un mayor precio medio de alquiler para apartamentos enteros?

¿Cual es su precio medio?

Ayuda: Usa la función aggregate `aggregate(.~colname,df,mean,na.rm=TRUE)`

¿Cuantos apartamentos hay en cada uno de esos barrios?

Mostrar una dataframe con el nombre del barrio, el precio y el número de entradas.

Ayuda: Podeis crear un nuevo dataframe con las columnas "Barrio" y "Freq" que contenga el número de entradas en cada barrio y hacer un merge con el dataframe del punto anterior.

Partiendo del dataframe anterior, muestra los 5 barrios con mayor precio, pero que tengan más de 100 entradas de alquiler.

Dibuja el diagrama de densidad de distribución de los diferentes precios

Calcula el tamaño medio, en metros cuadrados, para los 5 barrios anteriores y muestralo en el mismo dataframe junto con el precio y número de entradas

Dibuja el diagrama de densidad de distribución de los diferentes tamaños de apartamentos

Esta claro que las medias de cada uno de estos 5 barrios parecen ser diferentes, pero ¿son estadísticamente diferentes?
¿Que test habría que usar para comprobarlo?

Para únicamente los pisos de alquiler en el barrio de Sol:

``barrio_sol<-subset(airbnb_entire,Barrio=="Sol")``

Calcular un modelo lineal que combine alguna de estas variables:
* NumBanyos
* NumDormitorios
* MaxOcupantes
* MetrosCuadrados

Primero calculamos la correlación para ver como se relacionan estas variables entre sí.

Se observa que la correlación entre el número de dormitorios y los metros cuadrados es sorprendentemente baja.
¿Son de fiar esos números?

Mediante un histograma o curvas de densidad podemos descartar números que notienen sentido en el dataframe barrio_sol,
para tener una matriz de correlación que tenga mayor sentido.


Una vez que hayamos filtrado los datos correspondientes calcular el valor o la combinación de valores que mejor nos permite obtener el precio de un inmueble.

¿Que variable es más fiable para conocer el precio de un inmueble, el número de habitaciones o los metros cuadrados?

Responde con su correspondiente margen de error del 95%, ¿cuantos euros incrementa el precio del alquiler por cada metro cuadrado extra del piso?

Responde con su correspondiente margen de error del 95%, ¿cuantos euros incrementa el precio del alquiler por cada habitación?

¿Cual es la probabilidad de encontrar, en el barrio de Sol, un apartamento en alquiler con 3 dormitorios?
¿Cual es el margen de error de esa probabilidad?